modules/wsl-conf.nix

```diff
@@ -0,0 +1,90 @@
+{ lib, pkgs, config, ... }:
+with builtins; with lib; {
+
+  imports = [
+    (mkRenamedOptionModule [ "wsl" "automountPath" ] [ "wsl" "wslConf" "automount" "root" ])
+    (mkRenamedOptionModule [ "wsl" "automountOptions" ] [ "wsl" "wslConf" "automount" "options" ])
+  ];
+
+  # See https://learn.microsoft.com/en-us/windows/wsl/wsl-config#configuration-settings-for-wslconf for all options
+  options.wsl.wslConf = with types; {
+    automount = {
+      enabled = mkOption {
+        type = bool;
+        default = true;
+        description = "Automatically mount windows drives under ${config.wsl.wslConf.automount.root}";
+      };
+      mountFsTab = mkOption {
+        type = bool;
+        default = false;
+        description = "Mount entries from /etc/fstab through WSL. You should probably leave this on false, because systemd will mount those for you.";
+      };
+      root = mkOption {
+        type = str;
+        default = "/mnt/";
+        description = "The directory under which to mount windows drives.";
+      };
+      options = mkOption {
+        type = commas; # comma-separated strings
+        default = "metadata,uid=1000,gid=100";
+        description = "Comma-separated list of mount options that should be used for mounting windows drives.";
+      };
+    };
+    boot = {
+      command = mkOption {
+        type = str;
+        default = "";
```

Comment:
example would be great.

Reply:
This isn't really useful on NixOS, because we have systemd. The command is started using the root shell, so syschdemd would drop it to user privileges anyway. MS intends this to be used to start a daemon in the background (their example is starting docker)